sprokit/processes/core/merge_track_sets_process.cxx

```diff
@@ -0,0 +1,236 @@
+/*ckwg +29
+ * Copyright 2019 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "merge_track_sets_process.h"
+
+#include <vital/vital_types.h>
+#include <vital/types/object_track_set.h>
+#include <vital/util/string.h>
+
+#include <kwiver_type_traits.h>
+
+#include <sprokit/pipeline/process_exception.h>
+
+namespace kwiver
+{
+
+//-----------------------------------------------------------------------------
+// Private implementation class
+class merge_track_sets_process::priv
+{
+public:
+  priv() : highest_id( 0 ) {}
+  ~priv() {}
+
+  std::set< std::string > p_port_list;
+
+  vital::track_id_t highest_id;
+  std::set< vital::track_id_t > used_ids;
+  std::vector< std::map< vital::track_id_t, vital::track_id_t > > id_remapping;
+
+  void add_tracks_to_set( vital::track_set_sptr input, unsigned index,
+                          vital::track_set_sptr& output );
+};
+
+
+void
+merge_track_sets_process::priv
+::add_tracks_to_set( vital::track_set_sptr input, unsigned index,
+                     vital::track_set_sptr& output )
+{
+  if( !input )
+  {
+    return;
+  }
+
+  if( !output )
+  {
+    output = std::make_shared< vital::object_track_set >();
+  }
+
+  std::map< vital::track_id_t, vital::track_id_t >& mappings = id_remapping[index];
+
+  for( auto track_ptr : input->tracks() )
+  {
+    if( !track_ptr )
```

Comment:
Does this really happen? I would think having a null track pointer in a ``track_set`` is a failure in ``track_set``. Unless there is a reason for ``nullptr``s in the ``track_set``, they should not be allowed. Then users of the set would not have to do this test.